src/deterministic_vi/elbo_objective.jl

```diff
@@ -20,18 +20,17 @@ Returns:
 function calculate_source_pixel_brightness!{NumType <: Number}(
                     elbo_vars::ElboIntermediateVariables{NumType},
                     ea::ElboArgs{NumType},
-                    E_G_s::SensitiveFloat{CanonicalParams, NumType},
-                    var_G_s::SensitiveFloat{CanonicalParams, NumType},
-                    fs0m::SensitiveFloat{StarPosParams, NumType},
-                    fs1m::SensitiveFloat{GalaxyPosParams, NumType},
+                    E_G_s::SensitiveFloat{NumType},
+                    var_G_s::SensitiveFloat{NumType},
+                    fs0m::SensitiveFloat{NumType},
```

Comment:
In my opinion, by removing the parameter types here, not documenting the inputs more carefully, and not asserting the expected lengths of the SF parameters, you are opening the door for some real difficult-to-detect bugs in the future if someone gets their param types and arguments mixed up.  These functions will silently do very wrong things if the user passes in param types that are not the ones it expects.  In my opinion, the possibility of this kind of error is not worth simplifying the type.  But it's your call, ultimately.  Some length asserts at least would be a good idea.